src/features/archaeologist/discovery.ts

```diff
@@ -74,4 +80,31 @@ export async function initialisePeerDiscovery(browserNode: Libp2p, setArchs: (ar
   browserNode.pubsub.subscribe(archEnvConfigTopic);
 
   return browserNode;
-}
\ No newline at end of file
+}
+
+
+export async function confirmArweaveTransaction(data: { arch: Archaeologist, arweaveTxId: string, unencryptedShardHash: string }) {
+  const { arch, arweaveTxId, unencryptedShardHash } = data;
+  try {
+    const archConnection = discoveredArchs[arch.address].connection;
+
+    if (!archConnection) throw new Error('No connection to archaeologist');
+
+    const outboundMsg = JSON.stringify({
+      arweaveTxId,
+      unencryptedShardHash,
+    });
+
+    const outboundStream = pushable({});
+    outboundStream.push(new TextEncoder().encode(outboundMsg));
+
+    const { stream } = await archConnection.newStream('/validate-arweave');
+
+    pipe(
+      outboundStream,
+      stream,
+    );
```

Comment:
I think eventually we will want to abstract out the setup of the different streams and pipes so they are defined earlier in the lifecycle of the browser->arch connections.

These outbound streams should be global (or at least easily available to lookup), so then anywhere we need to pipe into a stream we can have a reference the stream we need to pipe to.